api/views/pois.py

```diff
@@ -59,35 +59,37 @@ def create_poi():
     data = request.get_json()
 
     # Check that all required parameters exist
-    if not all(i in data for i in ('name', 'description', 'map_year', 'x_coord', 'y_coord')):
+    required_fields = ['name', 'description', 'map_year', 'x_coord', 'y_coord']
+    missing_fields = [field for field in required_fields if data.get(field) is None]
+    if len(missing_fields):
         return create_response(
             status=422,
-            message='POI not created; missing required input parameter(s)'
+            message='POI not created; missing {}'.format(', '.join(missing_fields))
         )
-    data['links'] = data['links'] if 'links' in data else []
-    data['media'] = data['media'] if 'media' in data else []
-    data['story_ids'] = data['story_ids'] if 'story_ids' in data else []
+    content_fields = ['links', 'media', 'story_ids']
+    missing_content = [field for field in content_fields if type(data.get(field)) is not list]
+    for field in missing_content:
+        data[field] = []
     # Assumes that data['date'] is a parsable string containing a date if it exists
-    data['date'] = parse(data['date']) if 'date' in data else date(data['map_year'], 1, 1)
-    for link in data['links']:
-        if 'link_url' not in link:
-            return create_response(
+    data['date'] = parse(data['date']) if type(data.get('date')) is str else date(data['map_year'], 1, 1)
+    link_urls = [link.get('link_url') for link in data['links']]
+    media_urls = [media.get('content_url') for media in data['media']]
+    missing_stories = [story_id for story_id in data['story_ids'] if Story.query.get(story_id) is None]
+    if None in link_urls:
+        return create_response(
                 status=422,
                 message='POI not created; missing link URL'
-            )
-    for media in data['media']:
-        if 'content_url' not in media:
-            return create_response(
+        )
+    if None in media_urls:
+        return create_response(
                 status=422,
                 message='POI not created; missing media URL'
-            )
-
-    for story_id in data['story_ids']:
-        if Story.query.get(story_id) is None:
-            return create_response(
+        )
+    if len(missing_stories):
+        return create_response(
                 status=422,
-                message='POI not created; story_id {} does not exist'.format(story_id)
-            )
+                message='POI not created; missing story_id {}'.format(', '.join(missing_stories))

```

Comment:
i think the old message is a little more clear than "missing story_id ..." it can be mistaken for the request is missing and needs the specified story_ids (which is the opposite of what's actually going on). perhaps an even better message is something like `POI not created; supplied story_ids {} refer to nonexistent stories`